src/github.com/matrix-org/dendrite/common/log.go

```diff
@@ -15,26 +15,62 @@
 package common
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
-	"github.com/Sirupsen/logrus"
 	"github.com/matrix-org/dugong"
+	"github.com/mgutz/ansi"
+	"github.com/sirupsen/logrus"
 )
 
-type utcFormatter struct {
-	logrus.Formatter
+type dendriteFormatter struct {
+	logrus.TextFormatter
 }
 
-func (f utcFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+func (f dendriteFormatter) Format(entry *logrus.Entry) (format []byte, err error) {
 	entry.Time = entry.Time.UTC()
-	return f.Formatter.Format(entry)
+
+	if _, ok := entry.Data["prefix"]; ok {
+		prefix, ok := entry.Data["prefix"].(string)
+		if !ok {
+			return f.TextFormatter.Format(entry)
+		}
```

Comment:
You don't need to check if the key is in the map. `entry.Data["prefix"].(string)` is safe to even if the key is in the map.

This is because `entry.Data["prefix"]` will be `nil` and `prefix, ok := interface{}(nil).(string)` will be `"", false`.

https://play.golang.org/p/blNqSLNe7Y

I'd write this as:

```go
prefix, ok := entry.Data["prefix"].(string)
￼if !ok {
￼    return f.TextFormatter.Format(entry)
￼}
```

and unindent the rest of the function.